### ISSUE TITLE
google_app_engine_standard_app_version always re-creates resources instead of patching

### ISSUE THREAD

**Issue by pmarschik** [hashicorp/terraform-provider-google#8716]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment. If the issue is assigned to the "modular-magician" user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If the issue is assigned to a user, that user is claiming responsibility for the issue. If the issue is assigned to "hashibot", a community member has claimed the issue already.

### Description

Changing AppEngine scaling settings results in the resource being re-created (using a `POST https://appengine.googleapis.com/v1/{parent=apps/*/services/*}/versions` request) while a `PATCH https://appengine.googleapis.com/v1/{name=apps/*/services/*/versions/*}` would be sufficient if only certain fields are updated.

#### Use case
We use terraform to deploy AppEngine versions. In order for the rollout of a new version to be smooth for users of the service we would set `min_instances` of a new version to a higher value than normal, then switch the traffic via `google_app_engine_service_split_traffic` with `migrate_traffic = true` and after that is completed reduce `min_instances` to 0 again.
Since the version is always re-created this procedure cannot be used currently.

#### Possible solutions

From what I understand about https://github.com/GoogleCloudPlatform/magic-modules currently there would be two ways to implement that:

 * write `custom_create` code (tedious)
 * add a `NestedQuery` and split `automatic_scaling` into its own resource (breaks backwards compatibility)

### New or Affected Resource(s)

 * `google_app_engine_standard_app_version`
 * probably `google_app_engine_flexible_app_version` as well (didn't check this)

#### Using `custom_create`
```tf
resource "google_app_engine_standard_app_version" "default" {
   // ... standard setup omitted ...

  automatic_scaling {
    min_idle_instances      = var.min_idle_instances
    max_idle_instances      = var.max_idle_instances
    standard_scheduler_settings {
      min_instances = var.min_instances
      max_instances = var.max_instances
      target_cpu_utilization = var.target_cpu_utilization
      target_throughput_utilization = var.target_throughput_utilization
    }
  }
}

resource "google_app_engine_standard_app_version" "default" {
   // ... standard setup omitted ...

  manual_scaling {
    instances = var.instances
  }
}
```

#### Using `NestedQuery`

```tf
resource "google_app_engine_standard_app_version" "default" { /* ... */ }

resource "google_app_engine_standard_app_automatic_scaling" "default" {
    project = "some-project"
    service = "default"
    version = google_app_engine_standard_app_version.default
    min_idle_instances      = var.min_idle_instances
    max_idle_instances      = var.max_idle_instances
    standard_scheduler_settings {
      min_instances = var.min_instances
      max_instances = var.max_instances
      target_cpu_utilization = var.target_cpu_utilization
      target_throughput_utilization = var.target_throughput_utilization
    }
}
```

**Comment by c2thorn** [comment]
The API documentation lists the fields are supported in `PATCH`: https://cloud.google.com/appengine/docs/admin-api/reference/rest/v1/apps.services.versions/patch

@pmarschik for clarification: you want the provider to use `PATCH` when updating resources that are already in state? Your suggestion of `custom_create` implies that you are requesting the use of `PATCH` during resource creation (the resource is not yet in state). Just wanting to make sure.

**Comment by pmarschik** [comment]
@c2thorn yes I'd want the provider to use `PATCH` when updating resources already in state where supported, i.e. the fields mentioned in the API documentation. If any other fields were touched it should probably still do a `POST`. Sorry for the confusion.

**Comment by ggtisc** [comment]
It looks like this is solved, since after changing some arguments after running a `terraform plan` or `terraform apply` it returns an `update-in-place` action instead of `force-replacement` or `create-destroy`.

**Comment by melinath** [comment]
Closing this for now, as it doesn't seem like the resource recreates for all changes at this point. Please comment or open a new ticket if anything is unresolved here!
